parsel/selector.py

```diff
@@ -38,7 +39,13 @@ def _st(st):
 def create_root_node(text, parser_cls, base_url=None):
     """Create root node for text using given parser class.
     """
-    body = text.strip().encode('utf8') or b'<html/>'
+    body = None
+    # If text contains only comments like <!--hello-->
+    # etree.tostring will return None
+    if re.match(r'^(\s*<!--.*?-->\s*)*$', text, re.S):
```

Comment:
Hey @karlind , thanks for the patch and sorry it took a long time to review!

I'm afraid this regular expression will also match cases like: `u'<!-- comment --><body>something here</body><!-- another comment -->'`.

Also, please note that `create_root_node` function isn't exclusive to HTML, it has to support XML parsing as well.

I don't see a clear generic solution that would fit Parsel, I think right now it's best to handle these cases in the caller code.